lib/lab_manager/provider/vsphere.rb

```diff
@@ -129,9 +129,10 @@ def power_on
           task_result =  vs.vm_power_on(
             'instance_uuid' => instance_uuid
           )['task_state']
-          fail PowerOnEroor, "Power-on task finished in state: #{task_result}" unless
+          fail PowerOnError, "Power-on task finished in state: #{task_result}" unless
             task_result == 'success'
         end
+        compute.provider_data = vm_data(nil, vs: vs)
```

Comment:
instead of repeating line `compute.provider_data = vm_data(nil, vs: vs)` helper method whould be bettr:
`set_provider_data(nil, vs: vs)` or maybe call it  fetch_provider_data?